packages/test-ovm-full-node/src/stress-test.ts

```diff
@@ -0,0 +1,152 @@
+/* External Imports */
+import {getLogger, Logger, sleep} from '@eth-optimism/core-utils'
+import {JsonRpcProvider} from 'ethers/providers'
+
+const log: Logger = getLogger('stress-test')
+
+export interface Metrics {
+  bestMillis: number,
+  worstMillis: number
+  meanDurationMillis: number
+  medianDurationMillis: number
+  worstTenPercentMillis: number
+  worstFivePercentMillis: number
+  worstOnePercentMillis: number
+}
+
+export interface RequestResult {
+  index: number
+  request: string
+  response: string
+  responseDurationMillis: number
+  confirmationDurationMillis: number
+}
+
+export interface TestResult {
+  requestResults: RequestResult[]
+  requestCount: number
+  totalTimeMillis: number
+  requestsPerMilli: number
+  responseMetrics: Metrics
+  confirmMetrics: Metrics
+}
+
+/**
+ * Base class handling generic stress test functionality,
+ * leaving what is being tested to the implementor.
+ */
+export abstract class FullNodeStressTest {
+  private requestNumber
+
+  protected constructor(
+    protected readonly numberOfRequests,
+    protected readonly nodeUrl
+  ) {}
+
+  /**
+   * Runs a stress test with the configured number of requests and node URL using
+   * the contract deployed by the implementing class and the signed transactions
+   * created by the implementing class.
+   */
+  public async run(): Promise<TestResult> {
+    await this.deployContract()
+
+    this.requestNumber = 0
+
+    const signedTransactions: string[] = []
+    for (let i = 0; i < this.numberOfRequests; i++) {
+      signedTransactions.push(await this.getSignedTransaction())
+    }
+
+    const promises: Array<Promise<RequestResult>> = []
+    for (let i = 0; i < this.numberOfRequests; i++) {
+      promises.push(this.processSingleRequest(signedTransactions[i]))
+    }
+
+    const startTime: number = Date.now()
+    const requestResults: RequestResult[] = await Promise.all(promises)
```

Comment:
Could be somewhat interesting to allow for requests one after another, not just all at the same time.

Reply:
I assume you mean once the previous request has finished? If so, do you mean we got a response or it got mined? Being JS, all the requests are sent sequentially because it's single-threaded.